erdblick_app/app/tilesources.component.ts

```diff
@@ -0,0 +1,140 @@
+import {Component} from "@angular/core";
+import {ParametersService} from "./parameters.service";
+import {RightClickMenuService} from "./rightclickmenu.service";
+import {MapService} from "./map.service";
+import {SourceDataPanelComponent} from "./sourcedata.panel.component";
+import {InspectionService} from "./inspection.service";
+
+
+@Component({
+    selector: 'tilesources',
+    template: `
+        <p-dialog header="Inspect Tile Source Data" [(visible)]="menuService.tileSourceDataDialogVisible" [modal]="false"
+                  (onHide)="reset()" [style]="{'min-height': '14em', 'min-width': '36em'}">
+            <div *ngIf="loading" style="display:flex; justify-content: center">
+                <p-progressSpinner ariaLabel="loading"/>
+            </div>
+            <div *ngIf="!loading" class="tilesource-options">
+                <p *ngIf="errorString">{{ errorString }}</p>
+                <p-dropdown *ngIf="!errorString"
+                            [options]="tileIds"
+                            [(ngModel)]="selectedTileId"
+                            optionLabel="name"
+                            placeholder="Select a TileId"
+                            (ngModelChange)="onTileIdChange($event)"
+                            appendTo="body"/>
+                <p-dropdown *ngIf="!errorString"
+                            [options]="mapIds"
+                            [(ngModel)]="selectedMapId"
+                            [disabled]="!mapIds.length"
+                            optionLabel="name"
+                            [placeholder]="mapIds.length ? 'Select a MapId' : 'No associated maps found'"
+                            (ngModelChange)="onMapIdChange($event)"
+                            appendTo="body" />
+                <p-dropdown *ngIf="!errorString"
+                            [options]="sourceDataLayers" 
+                            [(ngModel)]="selectedSourceDataLayer" 
+                            [disabled]="!sourceDataLayers.length" 
+                            optionLabel="name" 
+                            placeholder="Select a SourceDataLayer"
+                            [placeholder]="mapIds.length ? 'Select a SourceDataLayer' : 'No associated source data layers found'"
+                            appendTo="body" />
+                <div style="display: flex; flex-direction: row; gap: 0.5em">
+                    <p-button *ngIf="!errorString" (click)="requestSourceData()" label="Load" icon="pi pi-check"></p-button>
+                    <p-button (click)="close()" label="Close" icon="pi pi-times"></p-button>
+                </div>
+            </div>
+        </p-dialog>
+    `,
+    styles: [``]
+})
+export class TileSourceDataComponent {
```

Comment:
Prefer e.g. `SourceDataLayerSelectionDialog` so as to not create the impression that this component is about presenting the source data. The file name triggers a similar ambiguity imo.